app/controllers/patients_controller.rb

```diff
@@ -466,18 +463,24 @@ def update_status
       end
     end
 
-    # Update patient and all group members
-    if params.permit(:apply_to_household)[:apply_to_household]
-      ([patient] + (current_user.get_patient(patient.responder_id)&.dependents || [])).uniq.each do |member|
-        update_monitoring_fields(member, params, patient[:id] == member[:id] ? :patient : :dependent, :group)
-      end
-      # Update patient and all group members in continuous exposure
-    elsif params.permit(:apply_to_household_cm_only)[:apply_to_household_cm_only] # update patient and group members only with continuous exposure on
-      ([patient] + (current_user.get_patient(patient.responder_id)&.dependents&.where(continuous_exposure: true) || [])).uniq.each do |member|
-        update_monitoring_fields(member, params, patient[:id] == member[:id] ? :patient : :dependent, :group_cm)
-      end
-    else # Update patient
-      update_monitoring_fields(patient, params, :patient, :none)
+    # Update patient
+    update_monitoring_fields(patient, params, :patient, :none)
+
+    # If not applying to household, return
+    return unless params.permit(:apply_to_household)[:apply_to_household] && params[:apply_to_household_ids].count.positive?
+
+    # If a household member has been removed, they should not be updated
+    current_household_member_ids = patient.household.where.not(id: patient.id).pluck(:id)
+    diff_household_array = params[:apply_to_household_ids] - current_household_member_ids
+    unless diff_household_array.empty?
+      error_message = 'Apply to household action failed: changes have been made to this household. Please refresh.'
+      render(json: { error: error_message }, status: :bad_request) && return
+    end
+
+    # Update selected group members if applying to household and ids are supplied
+    params[:apply_to_household_ids].each do |id|
```

Comment:
no because the `apply_to_household_ids` come from the apply to household table which never shows the patient that is currently being update.  it will never be in this list